src/app/navigation.service.ts

```diff
@@ -22,6 +33,29 @@ export class NavigationService {
     this.currentUrl = url;
   }
 
+  /**
+   * Gets the appropriate redirect path if any. Updates `possibleRedirectSignal` value as a side effect if non-null.
+   */
+  getPotentialRedirectPath(highlightCards: HighlightCard[]): string|null {
+    let redirectPath: string|null = null;
+    highlightCards.forEach(card => {
+      // CC 'Donate Today' or 'Donate Now' cards mean there should be a full page redirect (unless donor clicked
+      // the logo specifically to avoid that).
+      if (card.campaignFamily === 'christmasChallenge' && card.button.text.startsWith('Donate ')) {
+        redirectPath = card.button.href.pathname;
+      }
+    });
+
+    // At actual runtime in browser we expect this signal to be ready before redirect paths are checked because
+    // `setPossibleRedirectSignal()` is called in AppComponent's constructor, but this isn't guaranteed in all
+    // test contexts for now.
```

Comment:
IMO this is OK for the time being. The more substantial logic is unit tested in `NavigationService`; the signal use is simple and has been + will be further manually tested.